python/ray/serve/_private/utils.py

```diff
@@ -644,3 +645,82 @@ def get_capacity_adjusted_num_replicas(
         rounding=ROUND_HALF_UP
     )
     return max(1, int(rounded_adjusted_num_replicas))
+
+
+def calculate_desired_num_replicas(
```

Comment:
This is just moved form `autoscaling_policy.py`. No logical changes.

Reply:
This really doesn't belong in `utils.py`, please either move it to `autoscaling_policy.py` and underscore prefix it so it's clearly not a public API or keep it in `_private/autoscaling_policy.py`